src/Symfony/Component/Validator/Constraints/IbanValidator.php

```diff
@@ -64,8 +176,44 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        // ...start with a country code...
-        if (!ctype_alpha($canonicalized{0}) || !ctype_alpha($canonicalized{1})) {
+        // ...contain only digits and characters...
+        if (!ctype_alnum($canonicalized)) {
+            if ($this->context instanceof ExecutionContextInterface) {
+                $this->context->buildViolation($constraint->message)
+                    ->setParameter('{{ value }}', $this->formatValue($value))
+                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
+                    ->addViolation();
+            } else {
+                $this->buildViolation($constraint->message)
+                    ->setParameter('{{ value }}', $this->formatValue($value))
+                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
+                    ->addViolation();
+            }
+
+            return;
+        }
+
+        // ...contain uppercase characters only
+        if ($canonicalized !== strtoupper($canonicalized)) {
```

Comment:
it would change current behaviour;
https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Validator/Constraints/IbanValidator.php#L102

Wouldn't this be a task for a transformer? I'm not sure but some people don't like validator to change the value its testing. I'm in favor of removing the case test btw.